gemrb/GUIScripts/LUCommon.py

```diff
@@ -37,6 +37,45 @@ def GetNextLevelExp (Level, Class):
 	# if you change it, check that all callers can handle it
 	return 0
 
+def GetNextLevels (actor, Classes):
+	"""Returns the next level the PC will attain based on current XP"""
+
+	Level = [0]*3
+
+	# Used only for TNO, all other games divide XP/Number of classes
+	XPStat = [IE_XP, IE_XP_MAGE, IE_XP_THIEF]
+
+	NumClasses = len(Classes)
+
+	for i in range(NumClasses):
+		# Get the next level we will use to look up new stats from the 2da tables
+		if GUICommon.IsNamelessOne(actor):
+			#TNO is the only character in all games that tracks individual XP stats
+			Level[i] = GetNextLevelFromExp (GemRB.GetPlayerStat (actor, XPStat[i]), Classes[i])
+		else:
+			# In all other cases, XP cut exactly by number of classes
+			Level[i] = GetNextLevelFromExp (GemRB.GetPlayerStat (actor, IE_XP)/NumClasses, Classes[i])
+
+	return Level
+
+def GetLevelDiff (actor, Level):
+	"""Calculate the number of new levels based on current levels"""
+
+	LevelDiff = [0]*3
+	LevelStat = [IE_LEVEL, IE_LEVEL2, IE_LEVEL3]
+
```

Comment:
Have I understood and commented this behaviour correctly? The Dualswap thing gives me a bit of a headache so I am not sure.

Reply:
Kinda, you just reversed the example.
a FIGHTER that duals to MAGE will have FIGHTER as the original class, so FIGHTER_MAGE fits fine. But the other way around it would require a swap.

Reply:
So, a mage that duals to fighter will keep his mage level in IE_LEVEL, and fighter level in IE_LEVEL2?

Reply:
no, the current active class' level is always in IE_LEVEL.

Reply:
I think I know why I am getting confused...

![mc](https://user-images.githubusercontent.com/1923867/99198256-626c3f80-278f-11eb-9eb6-4e70bde70889.png)

 I guess if you start with a class on the left of the multi-class list, once you dual class the one on the right will be your primary class.. then the game will know which one is active. The old class is stored in the MC bits part of the cre file, then internally the game figures out which MC combination it should be to decide what abilities you can use... does that all sound right?

I've tested a few now and it seems to work ok, I just wanted to make sure I was reading it right. I need to fix some bugs and test BG2/IWD before I'm happy though.

Reply:
The game will know which one is active just by comparing levels. For DC chars it's always one or the other, regardless where in the process you are. "primary" is not a good word to use, since it's ambiguous.

When testing, do note that something is already broken for DCs: #819

Reply:
Ok, I've reworded the comment, hopefully it is a bit clearer now